mixer/adapter/kubernetes2/kubernetes.go

```diff
@@ -0,0 +1,401 @@
+		o.OriginServiceAccountName = p.Spec.ServiceAccountName
+	}
+	if len(p.Status.PodIP) > 0 {
+		o.OriginPodIp = net.ParseIP(p.Status.PodIP)
+	}
+	if len(p.Status.HostIP) > 0 {
+		o.OriginHostIp = net.ParseIP(p.Status.HostIP)
+	}
+	if app, found := p.Labels[params.PodLabelForService]; found {
+		n, err := canonicalName(app, p.Namespace, params.ClusterDomainName)
+		if err == nil {
+			o.OriginService = n
+		}
+	} else if app, found := p.Labels[params.PodLabelForIstioComponentService]; found {
+		n, err := canonicalName(app, p.Namespace, params.ClusterDomainName)
+		if err == nil {
+			o.OriginService = n
+		}
+	}
+}
+
+func fillDestinationAttrs(p *v1.Pod, o *ktmpl.Output, params *config.Params) {
+	if len(p.Labels) > 0 {
+		o.DestinationLabels = p.Labels
+	}
+	if len(p.Name) > 0 {
+		o.DestinationPodName = p.Name
+	}
+	if len(p.Namespace) > 0 {
+		o.DestinationNamespace = p.Namespace
+	}
+	if len(p.Spec.ServiceAccountName) > 0 {
+		o.DestinationServiceAccountName = p.Spec.ServiceAccountName
+	}
+	if len(p.Status.PodIP) > 0 {
+		o.DestinationPodIp = net.ParseIP(p.Status.PodIP)
+	}
+	if len(p.Status.HostIP) > 0 {
+		o.DestinationHostIp = net.ParseIP(p.Status.HostIP)
+	}
+	if app, found := p.Labels[params.PodLabelForService]; found {
+		n, err := canonicalName(app, p.Namespace, params.ClusterDomainName)
+		if err == nil {
+			o.DestinationService = n
+		}
+	} else if app, found := p.Labels[params.PodLabelForIstioComponentService]; found {
+		n, err := canonicalName(app, p.Namespace, params.ClusterDomainName)
+		if err == nil {
+			o.DestinationService = n
+		}
+	}
+}
+
+func fillSourceAttrs(p *v1.Pod, o *ktmpl.Output, params *config.Params) {
+	if len(p.Labels) > 0 {
+		o.SourceLabels = p.Labels
+	}
+	if len(p.Name) > 0 {
+		o.SourcePodName = p.Name
+	}
+	if len(p.Namespace) > 0 {
+		o.SourceNamespace = p.Namespace
+	}
+	if len(p.Spec.ServiceAccountName) > 0 {
+		o.SourceServiceAccountName = p.Spec.ServiceAccountName
+	}
+	if len(p.Status.PodIP) > 0 {
+		o.SourcePodIp = net.ParseIP(p.Status.PodIP)
+	}
+	if len(p.Status.HostIP) > 0 {
+		o.SourceHostIp = net.ParseIP(p.Status.HostIP)
+	}
+	if app, found := p.Labels[params.PodLabelForService]; found {
+		n, err := canonicalName(app, p.Namespace, params.ClusterDomainName)
+		if err == nil {
+			o.SourceService = n
+		}
+	} else if app, found := p.Labels[params.PodLabelForIstioComponentService]; found {
+		n, err := canonicalName(app, p.Namespace, params.ClusterDomainName)
+		if err == nil {
+			o.SourceService = n
+		}
+	}
+}
+
+func newCacheFromConfig(kubeconfigPath string, refreshDuration time.Duration, env adapter.Env) (cacheController, error) {
+	if env.Logger().VerbosityLevel(debugVerbosityLevel) {
+		env.Logger().Infof("getting kubeconfig from: %#v", kubeconfigPath)
+	}
+	config, err := getRESTConfig(kubeconfigPath)
+	if err != nil || config == nil {
+		return nil, fmt.Errorf("could not retrieve kubeconfig: %v", err)
+	}
+	if env.Logger().VerbosityLevel(debugVerbosityLevel) {
```

Comment:
Thoughout this source file, you don't need all these "if" checks around calling the logger. 

The only time you need to protect with an if like that, is if the actual call to the logger is expensive, such as if it requires calling fmt.Sprint before invoking the log method.